host.go

```
@@ -0,0 +1,195 @@
+				break loop
+			}
+			_, err = h.participant.ReceiveMessage(msg, false)
+		case <-ctx.Done():
+			return nil
+		}
+		if err != nil {
+			break loop
+		}
+	}
+	h.log.Errorf("gpbfthost exiting: %+v", err)
+	return err
+}
+
+//
+// internal APIs after
+//
+
+// Returns inputs to the next GPBFT instance.
+// These are:
+// - the EC chain to propose,
+// - the power table specifying the participants,
+// - the beacon value for generating tickets.
+// These will be used as input to a subsequent instance of the protocol.
+// The chain should be a suffix of the last chain notified to the host via
+// ReceiveDecision (or known to be final via some other channel).
+func (h *gpbfthost) GetChainForInstance(instance uint64) (gpbft.ECChain, error) {
+	// TODO: this is just a complete fake
+	ts := sim.NewTipSetGenerator(1)
+	chain, err := gpbft.NewChain(gpbft.TipSet{Epoch: 0, Key: ts.Sample()}, gpbft.TipSet{Epoch: 1, Key: ts.Sample()})
+	if err != nil {
+		h.log.Errorf("creating chain: %+v", err)
+		return nil, err
+	}
+
+	return chain, nil
+}
+
+func (h *gpbfthost) GetCommitteeForInstance(instance uint64) (*gpbft.PowerTable, []byte, error) {
+	table := gpbft.NewPowerTable()
+	err := table.Add(h.manifest.InitialPowerTable...)
+	if err != nil {
+		h.log.Errorf("creating powertable: %+v", err)
+		return nil, nil, err
+	}
+	return table, []byte{'A'}, nil
+}
+
+// Returns the network's name (for signature separation)
+func (h *gpbfthost) NetworkName() gpbft.NetworkName {
+	return h.manifest.NetworkName
+}
+
+// Sends a message to all other participants.
+// The message's sender must be one that the network interface can sign on behalf of.
+func (h *gpbfthost) Broadcast(msg *gpbft.GMessage) {
+	h.log.Info("broadcast")
+
+	var bw bytes.Buffer
+	err := msg.MarshalCBOR(&bw)
+	if err != nil {
+		h.log.Errorf("marshalling GMessage: %+v", err)
+	}
+	err = h.client.BroadcastMessage(h.runningCtx, bw.Bytes())
+	if err != nil {
+		h.log.Errorf("broadcasting GMessage: %+v", err)
```

I'd return the error to GPBFT and let it decide how to handle this.